tt_metal/llrt/tt_cluster.hpp

```diff
@@ -182,7 +182,7 @@ class Cluster {
     //       set_internal_routing_info_for_ethernet_cores(false);
     //       CloseDevice(0)
     //       CloseDevice(1)
-    void set_internal_routing_info_for_ethernet_cores(bool enable_internal_routing) const;
+    void set_internal_routing_info_for_ethernet_cores(bool enable_internal_routing, std::vector<chip_id_t> target_mmio_devices = {}) const;
```

Comment:
const ref

Reply:
I'm modifying it in the function 🐒 , oops. I'll make change

Reply:
`set_internal_routing_info_for_ethernet_cores` was writing to all chips to enable the erisc FW. But this is bugged when you open one set of mmio devices, because the next time you open the other mmio devices, the enable bit is already true before loading FW. Led to random ND timing sensitive hangs.